src/algebra/lie/subalgebra.lean

```diff
@@ -49,10 +49,20 @@ instance : has_zero (lie_subalgebra R L) :=
 
 instance : inhabited (lie_subalgebra R L) := ⟨0⟩
 instance : has_coe (lie_subalgebra R L) (submodule R L) := ⟨lie_subalgebra.to_submodule⟩
-instance : has_mem L (lie_subalgebra R L) := ⟨λ x L', x ∈ (L' : set L)⟩
 
 namespace lie_subalgebra
 
+open neg_mem_class
+
+instance : set_like (lie_subalgebra R L) L :=
+{ coe := λ L', L',
+  coe_injective' := λ L' L'' h, by { rcases L' with ⟨⟨⟩⟩, rcases L'' with ⟨⟨⟩⟩, congr' } }
+
+instance : add_subgroup_class (lie_subalgebra R L) L :=
+{ add_mem := λ L', L'.add_mem',
```

Comment:
It's longer as text, but a shorter term - assuming it works:
```suggestion
{ add_mem := lie_subalgebra.add_mem',
```

Reply:
This precise kind of changes is what breaks when you extend structures the new way. I'd advocate to keep the more robust proof to avoid later annoyance when copy-pasting.